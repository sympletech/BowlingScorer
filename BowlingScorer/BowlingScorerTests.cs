using ApprovalTests;
using ApprovalTests.Reporters;

using NUnit.Framework;

namespace BowlingScorer
{
    [TestFixture]
    [UseReporter(typeof(DiffReporter))]
    public class BowlingScorerTests
    {
        [Test]
        public void GutterBallTest()
        {
            // arrange
            var game = new Game();

            // act
            for (var i = 0; i < 20; i++)
                game.Roll(0);

            // assert
            Assert.AreEqual(0, game.Score);
        }

        [Test]
        public void Roll_a_Five()
        {
            // arrange
            var game = new Game();

            // act
            game.Roll(5);

            // assert
            Assert.AreEqual(5, game.Score);
        }

        [Test]
        public void Roll_Three_and_Four()
        {
            // arrange
            var game = new Game();

            // act
            game.Roll(3);
            game.Roll(4);

            // assert
            Approvals.Verify(game);
        }

        [Test]
        public void Roll_Three_Times()
        {
            // arrange
            var game = new Game();

            // act
            game.Roll(3);
            game.Roll(4);
            game.Roll(5);

            // assert
            Approvals.Verify(game);
        }

        [Test]
        public void CompleteFrame()
        {
            var frame = new Frame(null);

            frame.Rolls.Add(1);
            Assert.IsFalse(frame.Complete);
            frame.Rolls.Add(1);
            Assert.IsTrue(frame.Complete);
        }
    }
}
