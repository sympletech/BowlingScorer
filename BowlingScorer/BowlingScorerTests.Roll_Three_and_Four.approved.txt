Score : 7
1 : Rolls : 3, 4 | 7
